src/vm/runners/cairo_runner.rs

```diff
@@ -1022,6 +1022,44 @@ impl CairoRunner {
 
         Ok(())
     }
+
+    pub fn read_return_values(&mut self, vm: &VirtualMachine) -> Result<(), RunnerError> {
+        if !self.run_ended {
+            return Err(RunnerError::FinalizeNoEndRun);
+        }
+        let mut pointer = vm.get_ap();
+        for builtin_name in self.program.builtins.iter().rev() {
+            let builtin_runner = vm
+                .builtin_runners
+                .iter()
+                .find(|(name, _builtin)| builtin_name == name);
```

Comment:
I think this is a `HashMap`, so just do `builtin_runners.get(name)`.

Reply:
builtin_runners is actually a vec<(string, builtin_runner)>, that is why an iterator is used